### PR TITLE
Transition Travis -> GH actions

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,43 +3,75 @@ on:
     branches:
       - master
       - stable
-name: Build
+      - actions-test
```
Comment (pmderodat) [inline .github/workflows/main.yml:6]: Remove this?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -3,43 +3,75 @@ on:
...
+      GPR_PROJECT_PATH: /tmp/ADALIB_DIR/share/gpr
     steps:
       - name: Get libadalang
         uses: actions/checkout@v2
+
+      - name: Show PR branch ref
+        run: |
+            echo ${{ github.head_ref }}
+            echo ${{ github.repository_owner }}
+            echo ${{ github.event.pull_request.head.label }}
```
Comment (pmderodat) [inline .github/workflows/main.yml:27]: Is this still relevant? (now that you managed to get the `gh-langkit-branch.py` script working)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -3,43 +3,75 @@ on:
...
+        with:
+          repository: AdaCore/langkit
+          path: langkit
+          fetch-depth: 0  # all branches
+
+      - name: Install python dependencies
+        run: |
+            pip install -r langkit/REQUIREMENTS.dev
+            pip install jsonschema
+            python langkit/setup.py install
```
Comment (pmderodat) [inline .github/workflows/main.yml:45]: I suggest to run `pip install ./langkit` instead

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -49,63 +81,89 @@ jobs:
...
+            python gnatcoll-bindings/iconv/setup.py \
+                build -j0 --prefix=$PREFIX --library-types=relocatable
+            python gnatcoll-bindings/iconv/setup.py install
+            python gnatcoll-bindings/gmp/setup.py build \
+                -j0 --prefix=$PREFIX --library-types=relocatable
+            python gnatcoll-bindings/gmp/setup.py install
+
+      - name: Build lkt_parse
+        run: |
+            set -x -e
```
Comment (pmderodat) [inline .github/workflows/main.yml:115]: Is this still needed?

File: .github/workflows/main.yml (hunk 3 above)
Comment (raph-amiard) [reply .github/workflows/main.yml:45]: ok will do

File: .github/workflows/osx-build.yml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          gprinstall --uninstall --prefix=$prefix langkit_support.gpr
+          rm -rf build
+          # The gprbuild losts -lgmp on MacOS X, provide it in .cgpr file:
+          gprconfig --batch -o /tmp/file.cgpr --config=c --config=ada
+          sed -i -e '/for Shared_Library_Minimum_Switches use/s/$/ ("-lgmp") \&/' /tmp/file.cgpr
+          cat /tmp/file.cgpr
+          ada/manage.py generate -P
+          ada/manage.py --library-types relocatable build --build-mode ${BUILD:-prod} --gargs=--config=/tmp/file.cgpr
+          ada/manage.py --library-types relocatable install --build-mode ${BUILD:-prod} $prefix
+          tar czf libadalang-`basename $GITHUB_REF`-osx${DEBUG:+-dbg}.tar.gz -C $prefix .
```
[the remark this answers is not in the document]
Comment (raph-amiard) [reply .github/workflows/osx-build.yml:95]: Ok I added what's necessary to the GH action (I don't want to keep the travis).

**Review by pmderodat — APPROVED** [review]

**Pull request merged**